cmd/fpga_admissionwebhook/README.md

```diff
@@ -138,13 +120,31 @@ $ ./scripts/webhook-deploy.sh --ca-bundle-path /var/run/kubernetes/server-ca.crt
 
 # Mappings
 
-Requested FPGA resources are translated to AF resources. For example,
-`fpga.intel.com/arria10.dcp1.1-nlb0` is translated to `fpga.intel.com/af-d8424dc4a4a3c413f89e433683f9040b`.
+For the following mapping
+
+```yaml
+apiVersion: fpga.intel.com/v1
+kind: AcceleratorFunction
+metadata:
+  name: arria10.dcp1.1-nlb0
+spec:
+  afuId: d8424dc4a4a3c413f89e433683f9040b
+  interfaceId: 9926ab6d6c925a68aabca7d84c545738
+  mode: preprogrammed
+```
 
-In orchestrated mode, `fpga.intel.com/arria10.dcp1.1-nlb0` gets translated to
-`fpga.intel.com/region-9926ab6d6c925a68aabca7d84c545738`, and, the corresponding AF IDs are set in
-environment variables for the container. The [FPGA CRI-O hook](../fpga_crihook/README.md)
-then loads the requested bitstream to a region before the container is started.
+requested FPGA resources are translated to AF resources. For example,
+`fpga.intel.com/arria10.dcp1.1-nlb0` is translated to
+`fpga.intel.com/9926ab6d6c925a68aabca7d84c54573d8424dc4a4a3c413f89e433683f9040b`.
+The first 31 characters of the resource name part (`9926ab6d6c925a68aabca7d84c54573`)
+is the first 31 characters of the region interface ID for Arria10 with DCP1.1
+firmware. The next 32 characters (`d8424dc4a4a3c413f89e433683f9040b`) is an accelerator function ID.
+
+The same mapping, but with its mode field set to `orchestrated`, translates
```

Comment:
nit: "orchestrated" might be not the best choice of the terminology here for cluster admin UX, but I don't have right now better word.  Also, it might be good idea to accept "pre-programmed" as alias to "programmed ?

Reply:
Mm.. dunno. "programmed" can be read as "just in time programmed" aka orchestrated.
Maybe "static" and "dynamic"?

Reply:
> Also, it might be good idea to accept "pre-programmed" as alias to "programmed"

(This meant to say accept both pre-programmed and preprogrammed?)

> Maybe "static" and "dynamic"

this would work, IMO, but it might be good to start using "programming mode" in documentation where we just say only "mode" today.

Reply:
I've checked the docs, there's no place where just "mode" is mentioned. But in fpga_plugin's README.md `af` and `preprogrammed` are used interchangeably. The same with `region` and `orchestrated`. And "orchestration programmed" is an alias for `orchestrated`.

Probably we could rename `pregrogrammed` and `af` to `static`; `orchestrated` and `region` - to `dynamic`; `regiondevel` - to just `devel`. Also I'd avoid accepting aliases.

Reply:
Should we keep using  'region' and 'af' modes in order not to break compatibility with the currently used plugin modes?

Reply:
Ok, scratch my renaming proposal.

Let's leave it as it is now. But we need somebody (@grahamwhaley?) with a user's perspective to remove `preprogrammed` and `orchestrated`  from the FPGA plugin's README. Or to formulate better the relationships between the plugin's mode and the mode of a mapping.

Reply:
Ah, sadly I'm not a 'real' fpga user - I just took the existing docs and made them more structured and coherent.... I'd suggest finding somebody who really uses or owns the FPGA tools/workflows or maybe correlate against the FPGA driver docs to try and find the normal or common phrasing.

Reply:
@rojkov It seems I missed where the main idea of this change is explained, i.e. the flexibility of the mapping. In my understanding the key part of the mapping can be arbitrary string and only admin decides what that would be. Is my understanding correct?

Reply:
@bart0sh Yes, that's correct. I've added https://github.com/intel/intel-device-plugins-for-kubernetes/pull/358/commits/03b815c3fd128a7c8be75a409b0a3f3b13fb14ae. Could you check if it's clearer now?

Reply:
Yes, it's correct now. Thank you.

Reply:
@rojkov As for mode name I think we can continue using 'af' and 'region' as we got rid of webhook mode names 'preprogrammed' and 'orchestrated'. Although webhook names look better in my opinion, using plugin names would help us to avoid possible user confusions.

Reply:
Alright, I'll stick to the plugin mode names everywhere. Will update the PR.

Reply:
good idea, +1